plugins/pastefromword/filter/default.js

```diff
@@ -363,31 +392,49 @@
 				'v:imagedata': remove,
 				// This is how IE8 presents images.
 				'v:shape': function( element ) {
+					var emptyShapeInsideGroup = element.parent && element.parent.name === 'v:group';
+
 					// In chrome a <v:shape> element may be followed by an <img> element with the same content.
-					var duplicate = false;
-					element.parent.getFirst( function( child ) {
-						if ( child.name == 'img' &&
+					// Remain for legacy comaptibility (#995, #1069).
+					if ( !element.attributes[ 'o:gfxdata' ] && !emptyShapeInsideGroup ) {
+						var duplicate = false,
+							child;
+
+						// Canvas leaves empty `v:shape`, which should not be converted into img tag.
+						// These empty `v:shape` contains 2 attributes which helps distinguish it.
+						child = element.getFirst( 'v:imagedata' );
```

Comment:
This can be assigned on initialization, 5 lines above. And then the comment will fit better, right after `if` which it describes.

Reply:
I prefer not to move comment after `if, as it might imply relation to `shapeTagging` method. All other `if` description in entire file seems to be described just before it.

Reply:
@msamsel I fully agree, my mistake here, what I meant was 

> And then the comment will fit better, right **before** if which it describes

By merging `var` initialization you achieved just what I was thinking about so it is ok 👍